src/main/java/core/basesyntax/service/impl/WriterServiceImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.WriterService;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class WriterServiceImpl implements WriterService {
+    @Override
+    public void createReportFile(String fruitBalanceReport) {
+        File file = new File("src/main/resources/report.txt");
+        try (FileWriter fileWriter = new FileWriter(file)) {
+            fileWriter.write(fruitBalanceReport);
+        } catch (IOException e) {
+            throw new RuntimeException(e);

```

Comment:
Provide a more specific error message when throwing a RuntimeException, e.g., 'Failed to write report to the file: ' + e.getMessage()'.